src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml

```diff
@@ -61,5 +61,13 @@
 
             <tag name="messenger.transport_factory" />
         </service>
+
+        <service id="messenger.transport.doctrine.factory" class="Symfony\Component\Messenger\Transport\Doctrine\DoctrineTransportFactory">
+            <argument type="service" id="doctrine" on-invalid="ignore"/>
```

Comment:
This dep should be mandatory to make it work, isn't it? What about moving the full definition to `DoctrineBundle`?

Reply:
Oh I see like the transaction middleware. Adding this to my todo

Reply:
Maybe the entire code belongs to the Doctrine bridge rather than the Messenger component 🤔. WDYT ?

Reply:
IMHO the integration only.